workers/data_refinery_workers/processors/test_geo.py

```diff
@@ -87,8 +89,64 @@ def test_illumina_to_pcl(self):
         """ """
         from data_refinery_workers.processors import illumina
         job = prepare_illumina_job()
-        illumina.illumina_to_pcl(job.pk)
-        self.assertTrue(os.path.isfile('/home/user/data_store/raw/TEST/processed/GSE22427_non-normalized.PCL'))
+        final_context = illumina.illumina_to_pcl(job.pk)
+
+        for sample in final_context['samples']:
+            print(sample)
+            smashme = sample.get_most_recent_smashable_result_file()
+            print(smashme)
+            self.assertTrue(os.path.exists(smashme.absolute_file_path))
+            os.remove(smashme.absolute_file_path)
+
+    @tag("illumina")
+    def test_bad_illumina_detection(self):
+        """ With the wrong species, this will fail the platform detection threshold. """
+        from data_refinery_workers.processors import illumina
+        job = prepare_illumina_job('RATTUS_NORVEGICUS')
+        final_context = illumina.illumina_to_pcl(job.pk)
+        self.assertTrue(final_context['abort'])
+
+    @tag("illumina")
+    def test_good_detection(self):
+        """GSE54661 appears to be mislabled (illuminaHumanv4) on GEO. Shows our detector works. """
+        from data_refinery_workers.processors import illumina
+
+        pj = ProcessorJob()
+        pj.pipeline_applied = "ILLUMINA_TO_PCL"
+        pj.save()
+
+        og_file = OriginalFile()
+        og_file.source_filename = "ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE54nnn/GSE54661/suppl/GSE54661%5Fnon%5Fnormalized%2Etxt%2Egz"
+        og_file.filename = "GSE54661_non_normalized.txt"
+        og_file.absolute_file_path = "/home/user/data_store/raw/TEST/ILLUMINA/GSE54661_non_normalized.txt"
+        og_file.save()
+
+        assoc1 = ProcessorJobOriginalFileAssociation()
+        assoc1.original_file = og_file
+        assoc1.processor_job = pj
+        assoc1.save()
+
+        sample_names = [
+            "hypoxia_Signal",
+        ]
```

Comment:
Were there going to be more of these? I'm a bit confused why we're defining a list of length one to loop over.